releasenotes/notes/add-automatic-trace-uds-detection-7c7602c6a06f03d2.yaml

```
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Add automatic unix domain socket detection for traces. The expected path for the socket is `/var/run/datadog/apm.socket` which if exists, will be used instead of the previous http default, `localhost:8126`. To be used in conjunction with `apm_config.receiver_socket` in your `datadog.yaml` file, or the `DD_APM_RECEIVER_SOCKET` environment variable set on the Datadog agent.
```

```suggestion
    Add automatic unix domain socket detection for traces. The expected path for the socket is ``/var/run/datadog/apm.socket`` which if exists, will be used instead of the previous http default, ``localhost:8126``. To be used in conjunction with ``apm_config.receiver_socket`` in your ``datadog.yaml`` file, or the ``DD_APM_RECEIVER_SOCKET`` environment variable set on the Datadog agent.
```

this is RST, so needs double <code>``</code>.


Should we link to corp docs on setting up the agent? maybe keep what we have for archival reasons, but also link to corp docs where the unix socket setup stuff lives? (not a strong/blocking opinion, just a thought)

---

Do we have a doc that actually talks about unix socket setup stuff? The best I could find was the description here: https://docs.datadoghq.com/agent/docker/apm/?tab=linux#docker-apm-agent-environment-variables
If we have a better place in the corp docs than that then we should also link to it when we talk about the setting here in the Python corp docs: https://docs.datadoghq.com/tracing/setup_overview/setup/python/?tab=containers#configuration

---

I'll add what setting the path to a UDS looks in both docs though since I don't think it's very intuitive: `DD_TRACE_AGENT_URL=unix:///var/run/datadog/dsd.socket`

---

https://github.com/DataDog/documentation/pull/11758